### PR TITLE
Add dev.lynith.Speedtest

### PR DESCRIPTION
**Pull request by LynithDev**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [ ] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
    - Using the `--filesystem=~/.local/share` permission
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: dev.lynith.Speedtest.json [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--device=dri",
+        "--socket=wayland",
+        "--filesystem=~/.local/share/"
+    ],
+    "build-options" : {
+        "append-path" : "/usr/lib/sdk/rust-stable/bin",
+        "build-args" : [
+            "--share=network"
```
Comment (hfiguiere) [inline dev.lynith.Speedtest.json:21]: build must be offline. This is not allowed on the builders.

File: dev.lynith.Speedtest.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    ],
+    "modules" : [
+        {
+            "name" : "speedtest",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/LynithDev/speedtest-app"
```
Comment (hfiguiere) [inline dev.lynith.Speedtest.json:47]: a commit or tag is required.

File: dev.lynith.Speedtest.json [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                },
+                {
+                	"type": "archive",
+                	"url": "https://github.com/LynithDev/speedtest-app/releases/download/1.0.0/dev.lynith.Speedtest.zip"
```
Comment (hfiguiere) [inline dev.lynith.Speedtest.json:51]: a checksum is required.

But is it necessary if you pull the github repository?

File: dev.lynith.Speedtest.json [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        "org.freedesktop.Sdk.Extension.rust-stable"
+    ],
+    "command" : "speedtest",
+    "finish-args" : [
+        "--share=network",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--device=dri",
+        "--socket=wayland",
+        "--filesystem=~/.local/share/"
```
Comment (hfiguiere) [inline dev.lynith.Speedtest.json:16]: why do you need access to this?

Comment (LynithDev) [reply dev.lynith.Speedtest.json:16]: The program installs a *proprietary* binary from [Ookla's servers](https://www.speedtest.net/apps/cli) and stores it in `~/.local/share/speedtest/`

Comment (hfiguiere) [reply dev.lynith.Speedtest.json:16]: then it's not `filesystem` you need. It's `persist`

File: dev.lynith.Speedtest.json [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    ],
+    "build-options" : {
+        "append-path" : "/usr/lib/sdk/rust-stable/bin",
+        "build-args" : [],
```
Comment (hfiguiere) [inline dev.lynith.Speedtest.json:20]:
```suggestion
```

File: dev.lynith.Speedtest.json [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+        {
+            "name" : "speedtest",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/LynithDev/speedtest-app",
+                    "tag": "1.0.0"
+                }
```
Comment (hfiguiere) [inline dev.lynith.Speedtest.json:47]: you either need to add the crate sources here, see https://github.com/flatpak/flatpak-builder-tools/tree/master/cargo or use a tarball with vendord crates.

File: dev.lynith.Speedtest.json (hunk 4 above)
Comment (hfiguiere) [reply dev.lynith.Speedtest.json:16]: also does it handle architecture properly when downloading?

Comment (LynithDev) [reply dev.lynith.Speedtest.json:16]: > also does it handle architecture properly when downloading?

Yes

File: dev.lynith.Speedtest.json [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+        "org.freedesktop.Sdk.Extension.rust-stable"
+    ],
+    "command" : "speedtest",
+    "finish-args" : [
+        "--share=network",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--device=dri",
+        "--socket=wayland",
+        "--persist=~/.local/share/speedtest/"
```
Comment (hfiguiere) [inline dev.lynith.Speedtest.json:16]:
```suggestion
        "--persist=.local/share/speedtest/"
```

>        --persist=FILENAME
>           If the application doesn't have access to the real homedir, make the (homedir-relative) path FILENAME a bind mount to the corresponding path in
>           the per-application directory, allowing that location to be used for persistent data. This overrides to the Context section from the
>           application metadata. This option can be used multiple times.

**Comment by hfiguiere** [comment]
bot, build dev.lynith.Speedtest

File: dev.lynith.Speedtest.json [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+        "org.freedesktop.Sdk.Extension.rust-stable"
+    ],
+    "command" : "speedtest",
+    "finish-args" : [
+        "--share=network",
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--device=dri",
+        "--socket=wayland",
+        "--persist=.local/share/speedtest/"
```
Comment (hfiguiere) [inline dev.lynith.Speedtest.json:16]: now are you sure it downloads it in this directory instead of using the XDG standard?

**Pull request closed without merging**